src/components/DealInvestments.tsx

```diff
@@ -12,6 +16,32 @@ interface DealInvestmentsProps {
 export const DealInvestments: FunctionComponent<DealInvestmentsProps> = ({ className, deal }) => {
 	const intl = useIntl();
 	className = classNames([className, "space-y-6"]);
+	const [tranches, setTranches] = useState<Tranche[]>([]);
+	const [userTrancheBalances, setUserTrancheBalances] = useState<{ index: TokenAmount }>();
+	const { publicKey } = useWallet();
+
+	useEffect(() => {
+		const getInvestedTranches = async () => {
+			const balances = {} as { index: TokenAmount };
+			const tranches = await asyncFilter(deal.tranches.tranches, async (tranche) => {
+				try {
+					const userTrancheBalance = await tranche.userTrancheBalance(publicKey);
```

Comment:
if we are doing it for every tranche and it's still a separate call for every tranche, I prefer having this logic just be part of the TrancheInvestment component

Reply:
So just return null then in the TrancheInvestment component if there is no userTrancheBalance?

Reply:
So just return null then in the TrancheInvestment component if there is no userTrancheBalance?